ui/packages/catalog-ui-search/src/main/webapp/component/query-template-sharing/query-template-sharing.view.js

```diff
@@ -0,0 +1,214 @@
+            reset: 'render',
+            sync: 'cleanup'
+        },
+        regions: {
+            byEmail: '.template-sharing-by-email',
+            byRole: '.template-sharing-by-role'
+        },
+        events: {
+            'click .save': 'save',
+            'click .reset': 'render' // resetting can be done by re-rendering and flushing the state
+        },
+        serializeData: function () {
+            return {
+                link: window.location.href + '/' + this.model.get('id')
+            }
+        },
+        getSharingByEmail: function () {
+            return this.options.permissions.accessIndividuals != undefined ? 
+                this.options.permissions.accessIndividuals
+                .map(function (email) {
+                    return { value: email }
+                }) : [];
+        },
+        getSharingByRole: function () {
+            var view = this;
+
+            var roles = this.options.permissions.accessGroups != undefined ? 
+                this.options.permissions.accessGroups
+                .map(function (group) {
+                    return group;
+                }) : [];
+
+            return user.get('user').get('roles').map(function (role) {
+                return {
+                    attribute: 'role',
+                    action: (roles.indexOf(role) === -1) ? 'none' : 'access',
+                    value: role
+                };
+            });
+        },
+        onRender: function () {
+            this.collection = new Backbone.Collection(this.getSharingByRole());
+            this.emailCollection = new Backbone.Collection(this.getSharingByEmail());
+
+            this.byEmail.show(new EmailSharingEditor({
+                collection: this.emailCollection
+            }));
+
+            this.byRole.show(new RoleSharingEditor({
+                collection: this.collection
+            }));
+        },
+        save: function () {
+
+            var view = this;
+
+            Loading.beginLoading(view);
+
+            var emailList = this.emailCollection.map(function(email) {
+                return email.get('value')
+            });
+
+            var roleList = this.collection.chain()
+                .filter(function (role) {
+                    return role.get('action') !== 'none';
+                }).map(function (role) {
+                    return role.get('value');
+                }).value();
+
+            var templatePerms = {
+                    'security.access-individuals': emailList,
+                    'security.access-groups': roleList 
+            }
+
+            this.updateSharingPermissions(templatePerms)
+        },
+        updateSharingPermissions: function(templatePerms) {
+            var _this = this;
+            console.log(this.model.attributes)
+
+            $.ajax({
+                //TODO: This URL needs to be a constant pulled from the model?
+                url: "/search/catalog/internal/sharing/".concat(_this.options.modelId),
```

Comment:
Where should I grab this constant URL from? Seems bad to hardcode here

Reply:
I don't think it's terrible. The big positive is that if I do a `Find in Path` for `/internal/sharing` it will take me right to the component that makes the AJAX call.

Reply:
We have hard-coded URLs in the admin ui components although I'm not sure what we _should_ be doing.